src/main/java/bio/terra/workspace/mdc/MdcLogEnhancerFilter.java

```diff
@@ -0,0 +1,35 @@
+package bio.terra.workspace.mdc;
+
+import java.io.IOException;
+import java.util.concurrent.ThreadLocalRandom;
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import org.hashids.Hashids;
+import org.slf4j.MDC;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MdcLogEnhancerFilter implements Filter {
+
+  @Override
+  public void doFilter(
+      ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+      throws IOException, ServletException {
+    HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
+
+    String requestId =
+        ((requestId = httpRequest.getHeader("Request-Id")) != null)
+            ? requestId
+            : generateRequestId();
+
+    MDC.put("requestId", requestId + " ");
```

Comment:
I added a space after the `requestId` as extra padding because when testing locally it was bumping up against the log-level in the message i.e. it was producing messages that looked like `pn3N7J3MERROR`. That may have just been formatting in my terminal not properly showing a tab or something. If there are objects I can remove.